developer_guides/rimage/ipc4_extended_module_config.rst

```diff
@@ -0,0 +1,420 @@
+  * ``sample_rates``
+  * ``sample_sizes``
+  * ``sample_containers``
+  * ``channel_cfg``
+
+Values for above properties are available in common binmap file:
+``<repo_dir>/FW/intel_common/module_binmaps/dsp_fw_common.binmap``
+
+in respectively:
+  * ``stream_type <type>``
+  * ``sample_rates <list_of_supported_sample_rates>``
+  * ``sample_sizes <list_of_supported_sample_sizes>``
+  * ``sample_containers <list_of_supported_sample_containers>``
+  * ``channel_cfg <list_of_supported_channel_configurations>``
+
+.. code-block:: bash
+
+  enum pin {
+   in = 0, //input
+   out,  // output
+  }
+
+  enum stream_type {
+    pcm = 0,
+    mp3,
+  };
+
+  enum sample_rates {
+     8k,
+     11.5k,
+     12k,
+     16k,
+     18.9k,
+     22.05k,
+     24k,
+     32k,
+     37.8k,
+     44.1k,
+     48k,
+     64k,
+     88.2k,
+     96k,
+     176.4k,
+     192k
+  };
+
+  enum sample_sizes {
+     sample_8b,
+     sample_16b,
+     sample_24b,
+     sample_32b,
+     sample_64b,
+  };
+
+  enum sample_containers {
+     container_8b ,
+     container_16b,
+     container_32b,
+     container_64b,
+  };
+
+  enum channel_configurations {
+    // FRONT_CENTER
+     ch_mono,
+
+    // FRONT_LEFT | BACK_LEFT
+     ch_dual_mono,
+
+    // FRONT_LEFT | FRONT_RIGHT
+     ch_stereo,
+
+    // FRONT_LEFT | FRONT_RIGHT | LOW_FREQUENCY
+     ch_2_1,
+
+    // FRONT_LEFT | FRONT_RIGHT | FRONT_CENTER
+     ch_3_0,
+
+    // FRONT_LEFT | FRONT_RIGHT | BACK_LEFT  | BACK_RIGHT
+     ch_quad,
+
+    // FRONT_LEFT | FRONT_RIGHT | FRONT_CENTER | BACK_CENTER
+     ch_surround,
+
+    // FRONT_LEFT | FRONT_RIGHT | FRONT_CENTER | LOW_FREQUENCY
+     ch_3_1,
+
+    // FRONT_LEFT | FRONT_RIGHT | FRONT_CENTER | BACK_LEFT | BACK_RIGHT
+     ch_5_0,
+
+    // FRONT_LEFT | FRONT_RIGHT | FRONT_CENTER | SIDE_LEFT | SIDE_RIGHT
+     ch_5_0_surround,
+
+    // FRONT_LEFT | FRONT_RIGHT | FRONT_CENTER | LOW_FREQUENCY | BACK_LEFT | BACK_RIGHT
+     ch_5_1,
+
+    // FRONT_LEFT | FRONT_RIGHT | FRONT_CENTER | LOW_FREQUENCY | SIDE_LEFT | SIDE_RIGHT
+     ch_5_1_surround,
+
+    // FRONT_LEFT | FRONT_RIGHT | FRONT_CENTER | BACK_LEFT | BACK_RIGHT | FRONT_LEFT_OF_CENTER | FRONT_RIGHT_OF_CENTER
+     ch_7_0,
+
+    // FRONT_LEFT | FRONT_RIGHT | FRONT_CENTER | BACK_LEFT | BACK_RIGHT | SIDE_LEFT | SIDE_RIGHT
+     ch_7_0_surround,
+
+    // FRONT_LEFT | FRONT_RIGHT | FRONT_CENTER | LOW_FREQUENCY | BACK_LEFT | BACK_RIGHT | FRONT_LEFT_OF_CENTER | FRONT_RIGHT_OF_CENTER
+     ch_7_1,
+
+    // FRONT_LEFT | FRONT_RIGHT | FRONT_CENTER | LOW_FREQUENCY | BACK_LEFT | BACK_RIGHT | SIDE_LEFT | SIDE_RIGHT
+     ch_7_1_surround,
+  };
+
+
+Build new module entry
+**********************
+
+All these module setting are defined in module_binmaps and mod_cfgs files in FW/portable/platform/platform_name/ if the module is a existing one in reference fw and just do simple conversion and copy it to module entry. If the module is a totally new one, everything needs to build according to above definitions.
```

Comment:
```suggestion
All module settings are defined in module_binmaps and mod_cfgs files in FW/portable/platform/platform_name/. If the module already exists in the reference fw, do a simple conversion and copy it to module entry. If the module is new, build everything according to the above definitions.
```

Reply:
This is a very wordy paragraph. Please review it to make sure my edits did not alter the technical context.